api/src/main/java/com/velocitypowered/api/event/command/CommandExecuteEvent.java

```diff
@@ -80,6 +103,34 @@ public String toString() {
         + '}';
   }
 
+  /**
+   * Represents the source of a command invocation.
+   */
+  public enum InvocationSource {
```

Comment:
This is less of a "source" and more of a "type".
```suggestion
  public enum InvocationType {
```